### DISCUSSION TITLE
Regressors warning in AutoARIMA

### DISCUSSION THREAD

**Discussion by Gesh96** [Nixtla/statsforecast#807] [Q&A]
Hello!

I have recently come across a really strange warning when using the AutoARIMA model. I am currently using version 1.5.0 of statsforecast and running it on Python 3.10. The warning appears as follows:

**:\Users\georgi.gulyashki\AppData\Local\Programs\Python\Python310\lib\site-packages\statsforecast\arima.py:1562: UserWarning: xreg not required by this model, ignoring the provided regressors
  warnings.warn(**
  
 In any case, xreg is not an argument that is passed anywhere at all by the user, so I am a bit confused. I am using external regressors, so I am a bit worried that they are being ignored indeed. Do you have any idea what could be the cause of this warning?

Thanks a lot in advance for the help!

**Comment by jmoralez** [comment]
Hey. Can you provide a small reproducible example? It's important to know how you're providing the regressors to train and forecast

**Reply by Gesh96** [reply]
Sure, here is a snip of the code (where **y** is a dataframe with some regressors):

```
    # Index needed when using the statsforecast library
    y["unique_id"] = 1

    # Create a Fourier terms features in order to capture the yearly seasonality in the data
    y = y.set_index(pd.PeriodIndex(y[sequence_column], freq='D'))
    
    for i in range(1,4):
        y[f'sin365_{i}'] = np.sin(2 * i * np.pi * y.index.dayofyear / 365.25)
        y[f'cos365_{i}'] = np.cos(2 * i * np.pi * y.index.dayofyear / 365.25)
    
    y = y.reset_index(drop=True)

    # Rename the sequence and target columns as to follow the conventions of the statsforecast library
    y = y.rename(columns={sequence_column: "ds", target_column: "y"})

    # Will forecast horizon days ahead
    y_train = y.iloc[:-horizon] 
    y_test = y.iloc[-horizon:]
    
    # Define the model
    autoARIMA = AutoARIMA(season_length=7)
    model = StatsForecast(df=y_train.set_index('unique_id'),
                          models=[autoARIMA],
                          freq='D', n_jobs=1) 
    
    # Dataframe with external regressors
    xtest_df = y_test.drop(columns=["y"])
    
    # Perform the forecast
    fourier_df = model.forecast(h=horizon, X_df=xtest_df.set_index('unique_id'))
    fourier_df = fourier_df.reset_index()
```

**Reply by jmoralez** [reply]
I don't see anything wrong with this. Could you please try running it in the latest version (1.7.3)? We added a couple of validations which might help (also you don't need to set the id as the index anymore).
